You can no longer kill yourself with a SORD

Buggy123 opened the pull request:
Its literally so shitty it does less than half the damage of a donksoft dart with the soft cap removed. No way in hell you could impale yourself with it.

Also adds a way for any object to prevent suicide; just return NODIE on suicide_act

Buggy123 committed:
You can no longer kill yourself with a SORD

YotaXP commented:
As it should be. :+1:

Incoming5643 commented:
Are you sure this actually prevents the sord from killing you? I think suicide has fallbacks for this sort of thing that still get you killed.

Buggy123 commented:
Good question, I should check that...

drynwyn commented:
My reading of the code indicates that you'd die of oxyloss damage if you tried to suicide with a sord. Even if you didn't, you'd still get your "suiciding" var set to one, and would therefore have the suicided description on examine.

GunHog commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -49,8 +49,8 @@
-	user.visible_message("<span class='suicide'>[user] is impaling \himself with the [src.name]! It looks like \he's trying to commit suicide.</span>")
-	return(BRUTELOSS)
+	user.visible_message("<span class='suicide'>[user] tries to impale \himself with the [src.name]... and isn't even cut! Wow, that's a shitty sword...</span>")
+	return
```
If you cannot prevent suicide_act() killing the user, you could consider a different fluff message, such as killing yourself from the pure shame of having your robust hands defiled by such an awful weapon.

Buggy123 commented:
I tested it, and yeah you just use a generic suicide message instead. I'm testing a small addition to suicide.dm to fix this.

KorPhaeron commented:
Just check the staff of healing. It prevents suicides

Buggy123 commented:
Sweet that'll work perfectly, thanks!

Buggy123 committed:
# This is a combination of 10 commits.
# The first commit's message is:
Fixes the SORD preventing suicide
# This is the 2nd commit message:

Fixes path on SORD
# This is the 3rd commit message:

Fixed it maybe?
# This is the 4th commit message:

Maybe fixes it? please?
# This is the 5th commit message:

Ok we're gonna have to handle this in the suicide verb
# This is the 6th commit message:

THERE, done!
# This is the 7th commit message:

I probably need to squash these commits
# This is the 8th commit message:

really need to squash this
# This is the 9th commit message:

squishy squashy
# This is the 10th commit message:

REALLY GOTTA SQUASH THIS SHIT FAM

KazeEspada commented:
But you can suicide with EVERYTHING.

Your spacemen can snap his own fucking neck!

Buggy123 commented:
Shit, i still gotta figure this out then. I guess i can add a exception to the suicide verb.

Jordie0608 commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -47,11 +47,11 @@
-
+	
+/obj/item/weapon/sord/handle_suicide() //Literally too shitty to kill yourself with
+	src.visible_message("<span class='suicide'>[src] tries to impale \himself with the SORD... and isn't even cut! Wow, that's a shitty sword...</span>")
```
src. is implicit, can be removed.

KorPhaeron commented on code/game/verbs/suicide.dm:
```diff
@@ -8,8 +8,12 @@
 	if(confirm == "Yes")
-		suiciding = 1
 		var/obj/item/held_item = get_active_hand()
+		if(istype(held_item, /obj/item/weapon/sord)) //I should really expand this so other items can easily prevent suicide as well...
```
Snowflake check. Bad.

Buggy123 replied:
Ok, fine. Lemme see if i can figure out to make a "can'tsuicide" var for weapons.

KorPhaeron replied:
Adding a new var for all weapons is also bad

Buggy123 replied:
okay i came up with something elegant.

RemieRichards replied:
Why is the sord not just returning something like -1 but labeled as #define SUICIDE_NO_DEATH -1
and this code reacting to said define.

Infinitely better than a snowflake check.

Buggy123 commented:
there. elegant and simple. If a item returns 0 on suicide_act, you can't kill yourself with it. no snowflake shit, works for everything, breaks nothing.

phil235 commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -47,11 +47,11 @@
-	user.visible_message("<span class='suicide'>[user] is impaling \himself with the [src.name]! It looks like \he's trying to commit suicide.</span>")
-	return(BRUTELOSS)
-
+	user.visible_message("<span class='suicide'>You try to impale yourself with the SORD... and isn't even cut! Wow, that's a shitty sword...</span>")
```
The first argument of visible_message() is what every viewers see, so you to use [user] instead of "You". Use the second argument to make a special message for the user.

Buggy123 committed:
Fixes SORD not-suicide message

tkdrg commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -47,11 +47,11 @@
 	hitsound = 'sound/weapons/bladeslice.ogg'
 	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
-
+	
```
these tabs are pretty random

tkdrg commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -47,11 +47,11 @@
-	return(BRUTELOSS)
-
+	user.visible_message("<span class='suicide'>[user] tries to impale \himself with the SORD... and isn't even cut! Wow, that's a shitty sword...</span>", "<span class='suicide'>You try to impale yourself with the SORD... and don't even leave a mark! Maybe you should try something else...</span>")
+	return(0)
```
you can just `return 0` here, without ()s

tkdrg commented on code/game/verbs/suicide.dm:
```diff
@@ -8,10 +8,11 @@
 			var/damagetype = held_item.suicide_act(src)
+			if(damagetype == 0)
+				return //can't kill yourself with that, try something else.
 			if(damagetype)
```
you can just replace the three above lines with
```
if(!damagetype)
	return
```
as long as you indent the block below one level down.

tkdrg commented on code/game/verbs/suicide.dm:
```diff
@@ -58,7 +59,7 @@
 
 		visible_message("<span class='danger'>[suicide_message]</span>", "<span class='userdanger'>[suicide_message]</span>")
-
+		suiciding = 1
```
You can't put `suiciding = 1` this low, otherwise it will only be set if the person suicides without holding anything. Either put it back where it was and set it back to `0` on the failure block above, or just set it to 1 up there as well

Buggy123 committed:
Fixes several problems

Buggy123 committed:
Fixes several problems

tkdrg commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -49,8 +49,8 @@
-	user.visible_message("<span class='suicide'>[user] is impaling \himself with the [src.name]! It looks like \he's trying to commit suicide.</span>")
-	return(BRUTELOSS)
+	user.visible_message("<span class='suicide'>[user] tries to impale \himself with the SORD... and isn't even cut! Wow, that's a shitty sword...</span>", "<span class='suicide'>You try to impale yourself with the SORD... and don't even leave a mark! Maybe you should try something else...</span>")
+	return(NODIE)
```
you have to define this somewhere in order to be able to use it

Buggy123 commented:
Octareenroom91's is 1000x better and pretty much does the same thing.
